emcc.py

```diff
@@ -369,12 +369,12 @@ def get_clang_flags(user_args):
   if settings.RELOCATABLE and '-fPIC' not in user_args:
     flags.append('-fPIC')
 
-  # We use default visiibilty=default in emscripten even though the upstream
-  # backend defaults visibility=hidden.  This matched the expectations of C/C++
-  # code in the wild which expects undecorated symbols to be exported to other
-  # DSO's by default.
-  if not any(a.startswith('-fvisibility') for a in user_args):
-    flags.append('-fvisibility=default')
+    # For relocatable code we default to visibility=default in emscripten even
+    # though the upstream backend defaults visibility=hidden.  This matches the
+    # expectations of C/C++ code in the wild which expects undecorated symbols
+    # to be exported to other DSO's by default.
+    if not any(a.startswith('-fvisibility') for a in user_args):
+      flags.append('-fvisibility=default')
```

Comment:
Re-reading this now, I realize we don't get here if `-fPIC` was passed, as the condition on line 369 checks that (and not just `RELOCATABLE`). Is that ok?

Reply:
Oops, yes this should be in `if settings.RELOCATABLE`

Reply:
Done